### PR TITLE
#185172106 A buyer should be able to view their cart

### PR DESCRIPTION
**Pull request by Pacific911**
#### What does this PR do?
- It helps the buyer to view the cart  whenever he/she clicks the view cart button or a button to view the cart 
- Displays all the products from the cart

#### How should this be manually tested?
     - Click on the icon of cart to check your products in the cart but once you are logged in 

1. Below is the endpoint working 
     - GET: cart
#### What are the relevant pivotal tracker/Trello stories?
#185172106

#### Screenshots (if appropriate)
screen

### CODE REVIEW

**Review by munezerobagira — COMMENTED** [review]
Can make sure that whenever user click on the cart while it is empty shows the message that the cart is empty

File: src/pages/viewCart/Cart.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/* eslint-disable react-hooks/exhaustive-deps */
```
Comment (munezerobagira) [inline src/pages/viewCart/Cart.tsx:1]: why this comments?

File: src/pages/viewCart/Cart.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import { useEffect } from 'react';
+import '../viewCart/cart.scss';
+import { useGetCart } from './redux/hooks';
+// import loader from '../../assets/spinner.svg';
```
Comment (munezerobagira) [inline src/pages/viewCart/Cart.tsx:5]: same here

File: src/pages/viewCart/__test__/cartSlice.test.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (munezerobagira) [inline src/pages/viewCart/__test__/cartSlice.test.tsx:1]: why this line

**Review by munezerobagira — CHANGES_REQUESTED** [review]

File: src/pages/viewCart/Cart.tsx (hunk 2 above)
Comment (Pacific911) [reply src/pages/viewCart/Cart.tsx]: solved 👍

File: src/pages/viewCart/Cart.tsx (hunk 1 above)
Comment (Pacific911) [reply src/pages/viewCart/Cart.tsx]: corrected 👍

File: src/pages/viewCart/Cart.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+function CartTable() {
+  const { handleGetCart, result } = useGetCart();
+  useEffect(() => {
+    handleGetCart();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+  return result.loading ? (
+    <div>loading</div>
+  ) : (
+    <div>
```
Comment (AdrineUWERA) [inline src/pages/viewCart/Cart.tsx:14]: Consider a scenario when the user has no products in their cart

**Review by AdrineUWERA — CHANGES_REQUESTED** [review]

File: src/pages/viewCart/Cart.tsx (hunk 4 above)
Comment (Pacific911) [reply src/pages/viewCart/Cart.tsx:14]: done

**Review by AdrineUWERA — APPROVED** [review]

**Review by princoo — APPROVED** [review]

**Review by Edcreation — CHANGES_REQUESTED** [review]
Good work. Cart Table could be more responsive.

**Review by Edcreation — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**